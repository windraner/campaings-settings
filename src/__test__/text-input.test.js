import React from 'react';
import TextInput from '../components/main-content-section/tab-block/common/text-input/text-input';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<TextInput />', () => {
  it('Check for className and text', () => {
    const wrapper = shallow(<TextInput title="test1" value="test2" field="test3" stateHandler={function() {}} />);
    expect(wrapper.find('.campaigns-settings__input-title').length).toBe(1);
    expect(wrapper.find('.campaigns-settings__input-title').text()).toBe('test1');
    expect(wrapper.find('.campaigns-settings__text-input').length).toBe(1);
    expect(wrapper.find('.campaigns-settings__text-input').props().value).toBe('test2');
  });
  it('Check for callback is called', () => {
    const onClickFunc = jest.fn();
    const wrapper = shallow(<TextInput title="test1" value="test2" field="test3" stateHandler={onClickFunc} />);
    const input = wrapper.find('.campaigns-settings__text-input');
    input.simulate('change', { target: { value: 'foo' }});
    expect(onClickFunc).toBeCalled();
    expect(onClickFunc.mock.calls).toEqual([['test3', 'foo']]);
  });
  it('Check for hint showing', () => {
    const wrapper = shallow(<TextInput title="test1" value="test2" field="test3" stateHandler={function() {}} required={true} />);
    expect(wrapper.find('.campaigns-settings__hint-input-error').length).toBe(0);
    wrapper.setProps({ value: '' });
    expect(wrapper.find('.campaigns-settings__hint-input-error').length).toBe(1);
    expect(wrapper.find('.campaigns-settings__hint-input-error').text()).toBe('Can not be empty');
  });
  it('matches the snapshot hide', () => {
    const tree = shallow(<TextInput title="test1" value="test2" field="test3" stateHandler={function() {}} />);
    expect(toJson(tree)).toMatchSnapshot();
    tree.setProps({ value: '', required: true });
    expect(toJson(tree)).toMatchSnapshot();
  });
});
