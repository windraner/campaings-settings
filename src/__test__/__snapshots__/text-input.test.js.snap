// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> matches the snapshot hide 1`] = `
<Fragment>
  <label
    className="campaigns-settings__input-title"
  >
    test1
  </label>
  <input
    className="campaigns-settings__text-input  d-block w-100"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
    value="test2"
  />
</Fragment>
`;

exports[`<TextInput /> matches the snapshot hide 2`] = `
<Fragment>
  <label
    className="campaigns-settings__input-title"
  >
    test1
  </label>
  <input
    className="campaigns-settings__text-input campaigns-settings__text-input-error d-block w-100"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
    value=""
  />
  <div
    className="campaigns-settings__hint-input-error d-flex align-items-center mt-2"
  >
    <span
      className="campaigns-settings__hint-input-error-icon rounded-circle position-relative mr-2"
    />
    <span>
      Can not be empty
    </span>
  </div>
</Fragment>
`;
